src/service/plugins/mousepad.js

```diff
@@ -191,7 +191,7 @@ var Plugin = GObject.registerClass({
 
                 // Special key (eg. non-printable ASCII)
                 } else if (input.specialKey && KeyMap.has(input.specialKey)) {
-                    keysym = KeyMap.get(input.specialKey);
+                    keysym = KeyMap.get(parseInt(input.specialKey));
                     this._input.pressKeys(keysym, modifiers);
```

Comment:
I think the error is calling `this._input.pressKeys()` instead of `this._input.pressreleaseKeysym()`. If `specialKey` were a string, it would never pass the call to `KeyMap.has(input.specialKey)`.

Reply:
This is weird. Installing the extension from [Gnome Extensions](https://extensions.gnome.org/extension/1319/gsconnect), the Slideshow does not work. However, after cloning the repo and building it works.
I assumed my fix worked because after building, it just works for some reason.

Is this an issue just on my computer or is this common?

Reply:
Then the issue has been resolved in a commit after the last version release?

Definitely not the mousepad.js file under this PR. The last commit for that was in 2021.

Reply:
Oh shit! The gnome extension website version is still at v50, whereas the Github release page shows v53. It needs updating. I'll try to use the Github release and comment.

Reply:
The v53 release does not work. But as @adam-husain said building from the repo works. The full screen and exit presentation buttons work as well.

Reply:
This might have been inadvertently fixed by f2798d7, but the above error should still be fixed I think.

Reply:
Okay I think from what i understand the current function simulates key press but not a release, and you're suggesting a function which would simulate a key press followed by a release. Is that how it works?

Reply:
Well right now it calls `this._input.pressKeys()` which takes a list of characters, a string. It should be using `this,_input.pressreleaseKeysym()` which takes a single key symbol (a code for the symbol of a character) and presses/releases it. Since `KeyMap` is a map from KDE Connect codes to keysyms, I think that's what it should have always been.

Reply:
That sounds more correct to me, as well. The idea that you'd need to use `parseInt()` on the `KeyMap.get()` **after** passing a `KeyMap.has()` check _without_ it just doesn't make sense, in terms of program logic.

(If the same `parseInt()` were added to the `KeyMap.has()` check, then I could potentially get behind that change. But without it? Something's clearly off.)

Reply:
Looking at the definition, https://github.com/GSConnect/gnome-shell-extension-gsconnect/blob/281a9c9c4cf156374bf651f0956ecfc04ba08771/src/service/components/input.js#L214

I agree.

```js
keysym = KeyMap.get(input.specialKey);
this._input.pressreleaseKeysym(keysym);
```
That would be the change required then.

Reply:
Hrm. Which we can't do as a suggestion. But assuming @adam-husain has enabled upstream committer pushes to the PR branch, we can do it on the command line. Since it sounds like everyone is in agreement (yes?), I'll attempt to push a commit with that change now.

Reply:
Didn't work, so I opened #1418 from my fork. I included @adam-husain's original commits in the branch, so attribution is preserved.

Reply:
So, that was the wrong definition of `.pressreleaseKeysym()`. The actual one for the _input_ component is here:

https://github.com/GSConnect/gnome-shell-extension-gsconnect/blob/281a9c9c4cf156374bf651f0956ecfc04ba08771/src/service/components/input.js#L214-L222

...But it suffers from the same problem as the other: It doesn't take a `modifiers` argument, so the modifier keys are potentially getting lost if we use `pressreleaseKeysym()`. (See #1418.)

Reply:
(Er, sorry, duh. You had the right definition all along, @bevsxyz . Sourcegraph led me down a wrong path.)